carbonplan_tonyear/core.py

```diff
@@ -0,0 +1,141 @@
+        a = [0.2173, 0.2240, 0.2824, 0.2763]
+        tau = [0, 394.4, 36.54, 4.304]
+        t_horizon = np.arange(1001)
+    elif curve_name == 'ipcc_2007':
+        # parameters from IPCC AR4 2007 (Chapter 2, page 213)
+        # https://www.ipcc.ch/site/assets/uploads/2018/02/ar4-wg1-chapter2-1.pdf
+        a = [0.217, 0.259, 0.338, 0.186]
+        tau = [0, 172.9, 18.51, 1.186]
+        t_horizon = np.arange(1001)
+    elif curve_name == 'ipcc_2000':
+        # parameters from IPCC LULUCF Special Report 2000 (Chapter 2.3.6.3, Footnote 4)
+        # https://archive.ipcc.ch/ipccreports/sres/land_use/index.php?idp=74
+        a = [0.175602, 0.137467, 0.18576, 0.242302, 0.258868]
+        tau = [0, 421.093, 70.5965, 21.42165, 3.41537]
+        t_horizon = np.arange(1001)
+    else:
+        raise ValueError('No IRF parameters by the name \'' + curve_name + '\'.')
+
+    irf_curve = [a[0]] * len(t_horizon)
+    for t in t_horizon:
+        for i in np.arange(1, len(a)):
+            irf_curve[t] = irf_curve[t] + (a[i] * np.exp(-t / tau[i]))
+    return irf_curve
+
+
+def get_discounted_curve(discount_rate, curve):
+    return [x / math.pow(1 + discount_rate, i) for i, x in enumerate(curve)]
+
+
+def print_benefit_report(method_output):
+    discount = str(round(method_output['parameters']['discount_rate'] * 100, 1))
+    delay = str(method_output['parameters']['delay'])
+    baseline_atm_cost = str(round(method_output['baseline_atm_cost'], 2))
+    benefit = str(round(method_output['benefit'], 2))
+    num_needed = str(round(method_output['num_for_equivalence'], 1))
+
+    print()
+    print('Discount rate: ' + discount + '%')
+    print('Delay: ' + delay + ' year(s)')
+    print('Baseline atmospheric cost: ' + baseline_atm_cost + ' ton-years')
+    print('Benefit from 1tCO2 with delay: ' + benefit + ' ton-years')
+    print('Number needed: ' + num_needed)
+    print()
+
+
+def calculate_tonyears(method, baseline, time_horizon, delay, discount_rate):
+    """This function calculates the benefit of a delayed emission according one
+    of three ton-year accounting methods.
+
+    Keyword arguments:
+        method – string specifying the ton-year accounting method (Moura-Costa: \'mc\',
+        Lashof: \'lashof\', or IPCC \'ipcc\')
+        baseline – array modeling the residence of an emission in the atmosphere over time,
+        i.e. a decay curve / impulse response function
+        time_horizon – integer specifying the period over which the impact of an emission
+        is considered (years)
+        delay – integer specifying the emission delay for which a ton-year benefit will be
+        calculated (years)
+        discount_rate – float specifying the discount rate to apply time preference to both
+        costs and benefits over the time horizon
+
+    Function returns a dictionary with:
+        parameters – key parameters used for the calculation
+        baseline – array modeling baseline emission curve (discounted if applicable)
```

Comment:
this is a great doc string. 

i had a question about the output value of `baseline`, wondering if you'd intended for it to be the discounted baseline vs the 'raw' baseline. this answered my question!!!